src/coreclr/vm/jitinterface.cpp

```diff
@@ -7865,6 +7865,13 @@ CorInfoInline CEEInfo::canInline (CORINFO_METHOD_HANDLE hCaller,
         goto exit;
     }
 
+    if (pOrigCallerModule->IsInliningDisabledByProfiler())
+    {        
+        result = INLINE_NEVER;
```

Comment:
Why are we moving this outside of the PROFILING_SUPPORTED guard?
Also, why the change from INLINE_FAIL to INLINE_NEVER?

Reply:
From @simonferquel's past comment I'm guessing the intent was to move it outside the CORProfilerPresent() check because its possible to disable optimizations and then disconnect the profiler. I think the right spot would be inside the PROFILING_SUPPORTED build check but prior to the CORProfilerPresent().